components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -196,6 +197,13 @@ class Session(
         }
     }
 
+    /**
+     * Set when a load request is received, indicating if the user was involved in the interaction.
+     */
+    var triggeredByUserInteraction: Boolean by Delegates.observable(false) {
+        _, _, new -> notifyObservers { onLoadRequest(this@Session, new) }
```

Comment:
nit: Technically we are a bit "misusing" `Session` to forward events happening in `Engine.Observer`. Usually this here is a callback to notify subscribers to `Session` that a value has changed - however here we are always triggering an `onLoadRequest`.

Reply:
👍